### PR TITLE
Implement and Add SideDrawerField for Formula Field

### PR DESCRIPTION
**Pull request by mnmt7**
This change resolves the issue #1218 by introducing a new SideDrawerField component specifically for the Formula field and making appropriate changes to the codebase to utilize this new component.

screenshot

/claim #1218

### CODE REVIEW

**Comment by htuerker** [comment]
Hello, @mnmt7! 

First of all, thanks for the contribution!

Overall is quite good, I couldn't recognize any issue. I'll just ask some features here regarding my future concerns and give you some instructions how to implement with ease. 

First of all, problem is you're re-calculate formula function in SideDrawer. Which is good, if you don't make heavy computations or, most importantly dependent on deterministic code block. Whether the result is indeterministic or we have heavy computations(for example an http request), we don't want to re-calculate again.

Here's a simple example scenario: 
https://user-images.githubusercontent.com/46192266/234530411-763d18f2-082b-4375-93af-4a2f98951c92.mov

By now, we don't want to persist any calculation in firestore but later we may want to make it an option for our users. So, the trivial solution is to extend `omitRowyFields` 
```typescript
// src/utils/table
  delete shallowClonedRow["_rowy_ref"];
  delete shallowClonedRow["_rowy_outOfOrder"];
  delete shallowClonedRow["_rowy_missingRequiredFields"];
  delete shallowClonedRow["_rowy_new"];
  // !!
  delete shallowClonedRow["_rowy_formulaValue"];
```

Whenever the result has changed in useFormula, `useUpdateFieldAtom` to update `_rowy_formulaValue`, but make sure you don't persist data. Basically, just interrupt the execution when the fieldName is `_rowy_formulaValue`;

Also, you should figure out one more thing, there might be multiple formula fields in the row, so maybe you should consider using regexp to catch something like: `_rowy_formulaValue_columnKey`.

**Review by htuerker — CHANGES_REQUESTED** [review]
First of all, problem is you're re-calculate formula function in SideDrawer. Which is good, if you don't make heavy computations or, most importantly, it's dependent on deterministic code block. Whether the result is indeterministic or we have heavy computations(for example an http request), we don't want to re-calculate again.

Here's a simple example scenario: 
https://user-images.githubusercontent.com/46192266/234530411-763d18f2-082b-4375-93af-4a2f98951c92.mov

By now, we don't want to persist any calculation in firestore but later we may want to make it an option for our users. So, the trivial solution is to extend `omitRowyFields` 
```typescript
// src/utils/table
  delete shallowClonedRow["_rowy_ref"];
  delete shallowClonedRow["_rowy_outOfOrder"];
  delete shallowClonedRow["_rowy_missingRequiredFields"];
  delete shallowClonedRow["_rowy_new"];
  // !!
  delete shallowClonedRow["_rowy_formulaValue"];
```

Whenever the result has changed in useFormula, `useUpdateFieldAtom` to update `_rowy_formulaValue`, but make sure you don't persist data. Basically, just interrupt the execution when the fieldName is `_rowy_formulaValue`. Then, use that value in SideDrawer instead of calculating. This will also help us to implement persistence support in future.

Also, you should figure out one more thing, there might be multiple formula fields in the row, so maybe you should consider using regexp to catch something like: `_rowy_formulaValue_columnKey`.

Please don't hesitate to reach out to me for further assistance. Feel free to put your own solution, I'd love to see a different solution in case you think anything better.

**Comment by mnmt7** [comment]
@htuerker I have implemented the changes. Please review when possible.

File: src/atoms/tableScope/rowActions.ts [hunk 1]
```diff
@@ -423,6 +425,14 @@ export const updateFieldAtom = atom(
...
       });
 
+      // TODO(han): Formula field persistence
+      // const config = find(tableColumnsOrdered, (c) => {
+      //  const [, key] = fieldName.split("_rowy_formulaValue_");
+      //  return c.key === key;
+      // });
+      // if(!config.persist) return;
+      if (fieldName.startsWith("_rowy_formulaValue")) return;
+
```
Comment (htuerker) [inline src/atoms/tableScope/rowActions.ts:435]: Hey @mnmt7, I've just added it, it's required to prevent unnecessary db updates.

**Review by htuerker — APPROVED** [review]

**Comment by htuerker** [comment]
Hey again, @mnmt7! Thanks for your contribution. LGTM 👍

**Pull request merged**